browser/ui/views/sidebar/sidebar_edit_item_bubble_delegate_view.cc

```diff
@@ -0,0 +1,172 @@
+      /* between_child_spacing = */ 4));
+  views::Label::CustomFont title_font = {
+      GetFont(13, gfx::Font::Weight::NORMAL)};
+  auto* title = title_part->AddChildView(std::make_unique<views::Label>(
+      brave_l10n::GetLocalizedResourceUTF16String(
+          IDS_SIDEBAR_EDIT_ITEM_BUBBLE_TITLE),
+      title_font));
+  title->SetEnabledColor(
+      color_provider->GetColor(kColorSidebarAddBubbleHeaderText));
+  title->SetAutoColorReadabilityEnabled(false);
+  title->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+
+  title_tf_ = title_part->AddChildView(std::make_unique<views::Textfield>());
+  title_tf_->SetController(this);
+  title_tf_->SetText(target_item_.title);
+  title_tf_->SelectAll(true);
+  title_tf_->SetAccessibleName(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_SIDEBAR_EDIT_ITEM_BUBBLE_AX_TITLE_EDITOR_LABLE));
+
+  auto* url_part = AddChildView(std::make_unique<views::View>());
+  url_part->SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kVertical, gfx::Insets(),
+      /* between_child_spacing = */ 4));
+  auto* url = url_part->AddChildView(std::make_unique<views::Label>(
+      brave_l10n::GetLocalizedResourceUTF16String(
+          IDS_SIDEBAR_EDIT_ITEM_BUBBLE_URL),
+      title_font));
+  url->SetEnabledColor(
+      color_provider->GetColor(kColorSidebarAddBubbleHeaderText));
+  url->SetAutoColorReadabilityEnabled(false);
+  url->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+
+  url_tf_ = url_part->AddChildView(std::make_unique<views::Textfield>());
+  url_tf_->SetController(this);
+  url_tf_->SetText(base::UTF8ToUTF16(target_item_.url.spec()));
+  url_tf_->SelectAll(true);
+  url_tf_->SetAccessibleName(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_SIDEBAR_EDIT_ITEM_BUBBLE_AX_URL_EDITOR_LABLE));
+
+  UpdateOKButtonEnabledState();
+}
+
+views::View* SidebarEditItemBubbleDelegateView::GetInitiallyFocusedView() {
+  DCHECK(title_tf_);
+  return title_tf_;
+}
+
+void SidebarEditItemBubbleDelegateView::AddedToWidget() {
+  AddChildViews();
+}
+
+void SidebarEditItemBubbleDelegateView::ContentsChanged(
+    views::Textfield* sender,
+    const std::u16string& new_contents) {
+  UpdateOKButtonEnabledState();
+}
+
+void SidebarEditItemBubbleDelegateView::UpdateItem() {
+  std::u16string new_title = title_tf_->GetText();
+  if (new_title.empty())
+    new_title = url_tf_->GetText();
+  GetSidebarService(browser_)->UpdateItem(target_item_.url,
+                                          GURL(url_tf_->GetText()),
+                                          target_item_.title, new_title);
+}
+
+void SidebarEditItemBubbleDelegateView::UpdateOKButtonEnabledState() {
+  DCHECK(title_tf_ && url_tf_);
+
+  // To update item, we need valid url.
+  bool ok_button_enabled = !url_tf_->GetText().empty();
+  if (ok_button_enabled && target_item_.url == GURL(url_tf_->GetText()) &&
+      target_item_.title == title_tf_->GetText()) {
+    ok_button_enabled = false;
+  }
```

Comment:
I think it's okay to pass empty check.

```suggestion
  // Update item only when url or title is changed
  const bool ok_button_enabled = target_item_.url != GURL(url_tf_->GetText()) ||
      target_item_.title != title_tf_->GetText()
```

Reply:
Changed to use original url when url input is empty.
If url is empty, we don't know which item's title should be updated when user set different title.